go-apps/meep-gis-engine/server/gis-engine.go

```diff
@@ -1106,6 +1107,252 @@ func geGetAssetData(w http.ResponseWriter, r *http.Request) {
+	if err != nil {
+		log.Error(err.Error())
+		return nil
+	}
+	return &obj
+}
+
+func geGetDistanceGeoDataByName(w http.ResponseWriter, r *http.Request) {
+	// Get asset name from request path parameters
+	vars := mux.Vars(r)
+	assetName := vars["assetName"]
+	log.Debug("Get Distance GeoData for asset: ", assetName)
+
+	// Make sure scenario is active
+	if ge.activeModel.GetScenarioName() == "" {
+		err := errors.New("No active scenario")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+
+	srcAsset := ge.assets[assetName]
+	if srcAsset == nil {
+		err := errors.New("Asset not in scenario")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	srcLongStr := ""
+	srcLatStr := ""
+	if srcAsset.geoData != nil {
+		srcPosition := srcAsset.geoData.position
+		srcPoint := convertJsonToPoint(srcPosition)
+		srcLongStr = strconv.FormatFloat(float64(srcPoint.Coordinates[0]), 'f', -1, 32)
+		srcLatStr = strconv.FormatFloat(float64(srcPoint.Coordinates[1]), 'f', -1, 32)
+	} else {
+		err := errors.New("Asset has no geo location")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Retrieve Distance parameters from request body
+	var distanceParam DistanceParameters
+	if r.Body == nil {
+		err := errors.New("Request body is missing")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&distanceParam)
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	dstLong := float32(0.0)
+	dstLat := float32(0.0)
+	dstLongStr := ""
+	dstLatStr := ""
+	if distanceParam.AssetName != "" {
+
+		// Find second asset in active scenario model
+		dstAsset := ge.assets[distanceParam.AssetName]
+		if dstAsset == nil {
+			err := errors.New("Destination asset not in scenario")
+			log.Error(err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		if dstAsset.geoData != nil {
+			dstPosition := dstAsset.geoData.position
+			dstPoint := convertJsonToPoint(dstPosition)
+			dstLong = dstPoint.Coordinates[0]
+			dstLat = dstPoint.Coordinates[1]
+			dstLongStr = strconv.FormatFloat(float64(dstPoint.Coordinates[0]), 'f', -1, 32)
+			dstLatStr = strconv.FormatFloat(float64(dstPoint.Coordinates[1]), 'f', -1, 32)
+		} else {
+			err := errors.New("Destination asset has no geo location")
+			log.Error(err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+	} else {
+		dstLong = distanceParam.Longitude
+		dstLat = distanceParam.Latitude
+		dstLongStr = strconv.FormatFloat(float64(distanceParam.Longitude), 'f', -1, 32)
+		dstLatStr = strconv.FormatFloat(float64(distanceParam.Latitude), 'f', -1, 32)
+	}
+
+	srcCoordinates := "(" + srcLongStr + " " + srcLatStr + ")"
+	dstCoordinates := "(" + dstLongStr + " " + dstLatStr + ")"
```

Comment:
Why convert these to strings? Why not leave them as GeoJSON point coordinates.
Perhaps the API should take a point instead of a lat/long as well, this would be in line with the existing GeoData format.

Reply:
gis-cache in redis is also storing only as long and lat rather than a json.. should all of them be a similar format to prevent conversions? I left it as is for now...

Reply:
We can leave as is for now. If we need to edit/add/use this API differently in the future we can adapt the API & microservices accordingly.

Reply:
ok